src/utils.ts

```diff
@@ -178,22 +183,13 @@ export function getEncodedToken(token: Token): string {
  */
 export function getEncodedTokenV4(token: Token): string {
 	const idMap: { [id: string]: Array<Proof> } = {};
-	let mint: string | undefined = undefined;
-	for (let i = 0; i < token.token.length; i++) {
-		if (!mint) {
-			mint = token.token[i].mint;
+	const mint = token.mint;
+	for (let i = 0; i < token.proofs.length; i++) {
+		const proof = token.proofs[i];
+		if (idMap[proof.id]) {
+			idMap[proof.id].push(proof);
 		} else {
-			if (mint !== token.token[i].mint) {
-				throw new Error('Multimint token can not be encoded as V4 token');
-			}
-		}
-		for (let j = 0; j < token.token[i].proofs.length; j++) {
-			const proof = token.token[i].proofs[j];
-			if (idMap[proof.id]) {
-				idMap[proof.id].push(proof);
-			} else {
-				idMap[proof.id] = [proof];
-			}
+			idMap[proof.id] = [proof];
```

Comment:
What do you think about adding a check to make sure the `proof.id` is hex? 

On boardwalk, I was receiving V3 tokens with Cashu V0 keyset Id's, then when I would try to send and encode as a V4 token the `hexToBytes(id)` would error and say something like "Error: expected hex". I only suggest adding a check so that there can be a more clear error message: "Cannot encode proofs with a Cashu V0 keyset ID into a V4 token".

Reply:
Yes I totally agree. I wanted to implement something like this in https://github.com/cashubtc/cashu-ts/pull/194 to keep things separated.